From 748a7233f5fe1eae86bcedc21c2e9b38b57eaa09 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sat, 21 Jan 2023 02:45:36 +0100
Subject: [PATCH 299/321] Revert "mm/hugetlb: convert
 hugetlbfs_pagecache_present() to folios"

This reverts commit 9a1dd9756d794deffe683da8380d76bcccc1fa32.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 mm/hugetlb.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/mm/hugetlb.c b/mm/hugetlb.c
index 3862dc1ec..7ddb55c96 100644
--- a/mm/hugetlb.c
+++ b/mm/hugetlb.c
@@ -5656,15 +5656,15 @@ static bool hugetlbfs_pagecache_present(struct hstate *h,
 {
 	struct address_space *mapping;
 	pgoff_t idx;
-	struct folio *folio;
+	struct page *page;
 
 	mapping = vma->vm_file->f_mapping;
 	idx = vma_hugecache_offset(h, vma, address);
 
-	folio = filemap_get_folio(mapping, idx);
-	if (folio)
-		folio_put(folio);
-	return folio != NULL;
+	page = find_get_page(mapping, idx);
+	if (page)
+		put_page(page);
+	return page != NULL;
 }
 
 int hugetlb_add_to_page_cache(struct page *page, struct address_space *mapping,
-- 
2.39.0.rc2.1.gbd5df96b79

